bitarray/_bitarray.c

```diff
@@ -372,15 +372,20 @@ bitwise(bitarrayobject *self, PyObject *arg, enum op_type oper)
 }
 
 /* set the bits from start to stop (excluding) in self to val */
-static void
+idx_t
 setrange(bitarrayobject *self, idx_t start, idx_t stop, int val)
 {
     idx_t i;
+    idx_t ret = 0;
 
     assert(0 <= start && start <= self->nbits);
-    assert(0 <= stop && stop <= self->nbits);
+    assert(0 <= stop  && stop  <= self->nbits);
     for (i = start; i < stop; i++)
+    {
+        ret++;
         setbit(self, i, val);
```

Comment:
Hmm, using memset does not actually seem to give a perf boost, at least not on my machine and the test parameters I checked.  Maybe the compiler is clever enough to do the right thing here?